stanford/quiz/views.py

```diff
@@ -751,6 +758,35 @@ def send_whatsapp_view(request):
 
     return redirect('dashboard')
 
+@login_required
+def sign_up_course(request):
+    """ Signs up a user for a course """
+    if request.method == 'POST':
+        access_code = request.POST.get('access_code').strip()
+        student = request.user.student
+        if student and access_code != "":
+            if not Course.objects.filter(code=access_code):
+                return redirect('dashboard') # TODO how to display error easily?
+            student.courses.add(Course.objects.get(code=access_code))
+
+    return redirect('dashboard') # TODO how to display success message?
+
+
+@login_required
+def remove_course(request):
+    """ Removes user from a course"""
+    if request.method == 'POST':
+        access_code = request.POST.get('access_code').strip()
+        student = request.user.student
+        if student and access_code != "":
+            if not Course.objects.filter(code=access_code).filter(students=student):
+                return redirect('dashboard') # TODO how to display error easily?
```

Comment:
when would this error be reached? aren't they only able to remove courses they're enrolled in?

Reply:
fragment from old way of deleting courses through course number, can just leave it in if we re implement that functionality